src/services/obs/backend.rs

```diff
@@ -0,0 +1,310 @@
+
+        info!("backend use root {}", root);
+
+        // Handle endpoint and bucket.
+        let auth_bucket = match &self.auth_bucket {
+            Some(auth_bucket) => Ok(auth_bucket.clone()),
+            None => Err(other(BackendError::new(
+                HashMap::from([("auth_bucket".to_string(), "".to_string())]),
+                anyhow!("auth_bucket is empty"),
+            ))),
+        }?;
+
+        debug!("backend use auth_bucket {}", &auth_bucket);
+
+        let bucket = match &self.bucket {
+            Some(bucket) => bucket.to_string(),
+            None => "".to_string(),
+        };
+        debug!("backend use bucket {}", &bucket);
+
+        if !bucket.is_empty() && bucket != auth_bucket {
+            return Err(other(BackendError::new(
+                HashMap::from([
+                    ("auth_bucket".to_string(), auth_bucket),
+                    ("bucket".to_string(), bucket.to_string()),
+                ]),
+                anyhow!("bucket is set but not equal to auth_bucket"),
+            )));
+        }
+
+        let endpoint = match &self.endpoint {
+            Some(endpoint) => Ok(endpoint.clone()),
+            None => Err(other(BackendError::new(
+                HashMap::from([("endpoint".to_string(), "".to_string())]),
+                anyhow!("endpoint is empty"),
+            ))),
+        }?;
+        debug!("backend use endpoint {}", &endpoint);
+
+        let context = HashMap::from([
+            ("bucket".to_string(), bucket.to_string()),
+            ("endpoint".to_string(), endpoint.to_string()),
+        ]);
+
+        let client = HttpClient::new();
+
+        let mut signer_builder = Signer::builder();
+        if let (Some(access_key), Some(secret_key)) = (&self.access_key, &self.secret_key) {
+            signer_builder.access_key(access_key).secret_key(secret_key);
+        }
+
+        // Bucket name must be set in CanonicalizedResource.
+        signer_builder.bucket(&auth_bucket);
+
+        let signer = signer_builder
+            .build()
+            .map_err(|e| other(BackendError::new(context, e)))?;
+
+        info!("backend build finished: {:?}", &self);
+        Ok(Backend {
+            client,
+            root,
+            endpoint,
+            signer: Arc::new(signer),
+            bucket,
+        })
+    }
+}
+
+// Backend for Huaweicloud OBS services.
+#[derive(Debug, Clone)]
+pub struct Backend {
+    client: HttpClient,
+    root: String,
+    endpoint: String,
+    signer: Arc<Signer>,
+    bucket: String,
+}
+
+impl Backend {
+    pub(crate) fn get_abs_path(&self, path: &str) -> String {
+        util::get_abs_path(&self.root, path)
+    }
+}
+
+#[async_trait]
+impl Accessor for Backend {
+    fn metadata(&self) -> AccessorMetadata {
+        let mut am = AccessorMetadata::default();
+        am.set_scheme(Scheme::OBS)
+            .set_root(&self.root)
+            .set_name(&self.bucket)
+            .set_capabilities(None);
```

Comment:
```suggestion
            .set_name(&self.bucket);
```

It's OK to ignore `set_capabilities` if no extra capabilities.